bob-common/src/mapper.rs

```diff
@@ -116,24 +122,26 @@ impl Virtual {
     }
 
     pub fn get_support_nodes(&self, key: BobKey, count: usize) -> Vec<&Node> {
+        debug_assert!(count < self.nodes.len());
         trace!("get target nodes for given key");
         let target_nodes = self.get_target_nodes_for_key(key);
         trace!("extract indexes of target nodes");
-        let mut target_indexes = target_nodes.iter().map(Node::index);
-        let len = target_indexes.size_hint().0;
-        debug!("iterator size lower bound: {}", len);
         trace!("nodes available: {}", self.nodes.len());
-        self.nodes
-            .iter()
-            .filter_map(|(id, node)| {
-                if target_indexes.all(|i| &i != id) {
-                    Some(node)
-                } else {
-                    None
+        let offset = self.support_nodes_offset.fetch_add(1, Ordering::Relaxed);
+        let mut support_nodes = Vec::with_capacity(count);
+        for (id, node) in self.nodes.iter().skip(offset % self.nodes.len()) {
+            if target_nodes.iter().all(|i| i.index() != *id) {
+                support_nodes.push(node)
```

Comment:
You should `break` if `support_nodes.len() >= count`

Reply:
Done.